### ISSUE TITLE
AutoMath module - add a `magnitude` function

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#23174]
In the discussion about the naming of `carg`/`phase` (#19006), it was pointed out that `arg`/`abs` and `phase`/`magnitude` are the typical term pairings.  Given that `arg` has other meanings in Chapel, we ended up deciding to rename `carg` to `phase` for clarity.  However, this means that we no longer support a full term pairing and are instead mixing and matching.

We don't think it would be appropriate to stop using `abs` for complex numbers, but it did seem beneficial to additionally provide a `magnitude` function, even though this would mean that there are two ways of accessing the functionality.  We would probably use the full name instead of `mag`, given that we anticipate this function being included in all programs by default (like `phase` and `abs` are, given that they seem like basic functionality for a complex number).

**Comment by lydia-duncan** [comment]
We don't anticipate adding this function before 2.0

**Comment by mppf** [comment]
I agree with the last comment; I don't personally see benefit from having `magnitude` just to pair with `phase`. I think `abs` is clear enough and has enough precedent in math for complex numbers. (But of course, in math, we write `abs` with vertical bars as in `|x|`; but IMO `abs` is an acceptable transliteration unless/until we are trying to make programming more like writing LaTex).

**Comment by damianmoz** [comment]
Michael's comment:
```
unless/until we are trying to make programming more like writing LaTex).
```
I think that was what the long-since departed competitor, Fortress, was all about. I much prefer Chapel. Besides it has been way more successful in the real world. Sorry if I digress too much.
